src/main/java/com/github/checkstyle/regression/module/ConfigGenerator.java

```diff
@@ -0,0 +1,64 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle.regression.module;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.github.checkstyle.regression.data.ConfigNode;
+import com.github.checkstyle.regression.data.GitChange;
+import com.github.checkstyle.regression.data.ImmutableConfigNode;
+
+/**
+ * Contains the logic of regression config generation.
+ * @author LuoLiangchen
+ */
+public final class ConfigGenerator {
+    /** Prevents instantiation. */
+    private ConfigGenerator() {
+    }
+
+    /**
+     * Generates the regression config nodes from a list of changes.
+     * @param changes the changes source
+     * @return the regression config nodes generated from the given changes
+     */
+    public static List<ConfigNode> generateConfigNodes(List<GitChange> changes) {
```

Comment:
The only duty of `configuration` is build the configuration file. There are no specifics on the type of data it should retrieve.
Someone has to build the configuration, and someone has to generate the list of different modules to build based on properties and such. It's not a concern who does which, but do we need an intermediate data type before the `direct data describing the XML`.

> I was thinking that configuration would only get direct data describing the XML.(

My reasoning against this is "what if we have to build a new component between module and configuration". Can this new component deal with the design structure of the final configuration?
Would it just want to see a single module as a whole?

We can hold this off for now, but I am just worried about if this structure will be good for https://github.com/checkstyle/regression-tool/issues/5 as we need to create multiple module configurations for each set of properties.
Which is why I think the answer to `How will you maintain list of properties and values to cycle through?` is an important question.